packages/core/features/src/connect.ts

```diff
@@ -37,7 +37,7 @@ export type ConnectFeature = StandardConnectFeature;
  *
  * @group Connect
  */
-export type StandardConnectVersion = '1.0.0';
+export type StandardConnectVersion = '1.1.0';
```

Comment:
I know we've talked about this before, but now that it's actually happening: how should we do this?

The problems are:

1. Because `features` is an object, there's no way for a wallet to provide two different versions of the same feature (ie. the keys – the name of the feature – would collide)
2. There's no way, at present, to call `connect()` and say ‘show me wallets that support this feature at this version.’

Possible solutions, all breaking, are:

1. Shape `features` like this:
    ```ts
    {
        'standard:connect': {
            '1.0.0': { connect(config) { /* ... */ } },
            '1.1.0': { connect(config) { /* ... */ } },
        },
    }
   ```
2. Add the version to the feature name
    ```ts
    {
        'standard:connect:1.0.0': { connect(config) { /* ... */ } },
        'standard:connect:1.1.0': { connect(config) { /* ... */ } },
    }
    ```

Both of these options could _kinda_ be backward compatible if we could convince wallets to do something like this:

```ts
function semverSort(a, b) { /* ... */ }
function getLatestFeature(versions) {
    const sortedVersions = Object.keys(feature).sort(semverSort);
    const latestVersion = sortedVersions[sortedVersions.length - 1];
    return versions[latestVersion];
}
// For option 1
features = {
    'standard:connect': new Proxy(
        {
            '1.0.0': { connect(config) { /* ... */ } },
            '1.1.0': { connect(config) { /* ... */ } },
        },
        {
            get(target, p, receiver) {
                const value = Reflect.get(target, p, receiver);
                if (value) {
                    return value;
                }
                return getLatestFeature(target);
            }
        }
    ),
}

// For option 2
function semverSort(a, b) { /* ... */ }
function getLatestFeature(features, featureName) {
    const sortedVersions = Object.keys(features)
        .filter(k => k.startsWith(`${featureName}:`))
        .map(k => k.replace(new RegExp(`^${featureName}:`), ''))
        .sort(semverSort);
    const latestVersion = sortedVersions[sortedVersions.length - 1];
    return features[`${featureName}:${latestVersion}`];
}
features = new Proxy(
    {
        'standard:connect:1.0.0': { connect(config) { /* ... */ } },
        'standard:connect:1.1.0': { connect(config) { /* ... */ } },
    },
    {
        get(target, p, receiver) {
            const value = Reflect.get(target, p, receiver);
            if (value) {
                return value;
            }
            const featureName = p.toString();
            return getLatestFeature(target, featureName);
        }
    }
}
```

That way, this code would continue to work:

```ts
// Gets the latest version for option 1
getLatestFeature(features['standard:connect']).connect();
features['standard:connect']['1.1.0'].connect();
// Gets the latest version for option 2
getLatestFeature(features, 'standard:connect').connect();
features['standard:connect:1.1.0'].connect();
// Gets the latest version in all cases
features['standard:connect'].connect();
```

Reply:
Yeah, I like this. Even without a Proxy, you could just do

```ts
features: {
    'standard:connect': {
        version: '1.0.0',
        connect () { /* ... */ },
        '1.0.0': {
            connect () { /* ... */ },
        }
        '1.1.0': {
            connect () { /* ... */ },
        }
    }
}
```

so this could be purely additive (in the case of option 1, which I like more than changing the name for substantially the same or backwards compatible API)

Reply:
We could also add a key like `versions: ['1.0.0', 1.1.0']` or

```
    versions: {
        '1.0.0': { /* ... */ },
        '1.1.0': { /* ... */ },
    }
```